src/sentry/testutils/helpers/features.py

```diff
@@ -141,7 +142,73 @@ def batch_features_override(
             yield
 
 
-with_feature = Feature
+def with_feature(names: str | Iterable[str] | dict[str, bool]):
+    """
+    Control whether a feature is enabled.
+
+    Can be used as:
+    - Context manager: with with_feature('feature-name'):
+    - Method decorator: @with_feature('feature-name')
+    - Class decorator: @with_feature('feature-name')
+    """
+
+    class FeatureContextManagerOrDecorator:
+        def __init__(self, feature_names):
+            self.feature_names = feature_names
+            self._context_manager = None
+
+        def __enter__(self):
+            self._context_manager = Feature(self.feature_names)
+            return self._context_manager.__enter__()
+
+        def __exit__(self, exc_type, exc_val, exc_tb):
+            if self._context_manager:
+                return self._context_manager.__exit__(exc_type, exc_val, exc_tb)
+
+        def __call__(self, func_or_cls):
+            # Check if we're decorating a class
+            if isinstance(func_or_cls, type):
+                # Apply feature flag to the entire class
+                # First, create pytest fixture for compatibility with pytest runs
+                feature_names = self.feature_names
+
+                def _feature_fixture(self: object) -> Generator[None]:
+                    with Feature(feature_names):
+                        yield
+
+                name = f"{_feature_fixture.__name__}[{feature_names}]"
+                _feature_fixture.__name__ = name
+                fixture = pytest.fixture(scope="class", autouse=True)(_feature_fixture)
+                setattr(func_or_cls, name, fixture)
+
+                # Additionally, wrap each test method directly so it works when called manually
+                for attr_name in dir(func_or_cls):
+                    if attr_name.startswith("test_"):
```

Comment:
This behavior seems a bit weird to me. Are we potentially not wrapping helper/setup/teardown methods? If so, it would be a bit unusual not to have feature flags enabled when the entire class has this decorator.